homeassistant/components/homekit_controller/sensor.py

```diff
@@ -203,17 +205,17 @@ class HomeKitHumiditySensor(HomeKitEntity, SensorEntity):
     _attr_device_class = SensorDeviceClass.HUMIDITY
     _attr_native_unit_of_measurement = PERCENTAGE
 
-    def get_characteristic_types(self):
+    def get_characteristic_types(self) -> list[str]:
         """Define the homekit characteristics the entity is tracking."""
         return [CharacteristicsTypes.RELATIVE_HUMIDITY_CURRENT]
 
     @property
-    def name(self):
+    def name(self) -> str:
         """Return the name of the device."""
         return f"{super().name} Humidity"
 
     @property
-    def native_value(self):
+    def native_value(self) -> StateType:
```

Comment:
```suggestion
    def native_value(self) -> float:
```

According to the HAP spec this should only be `float`

Is there ever a case where we would get data from a non-compliant device that we need to have looser typing?

Reply:
I mean I have 0 confidence for the device to not return a string or something. But I think if that happens we need to deal with it before it hits the state machine, and ideally in aiohomekit.

We can always soften the restriction again if it becomes a problem.

Right now `.value()` returns an Any (😱) so its all a bit moot.